### PR TITLE
Use eternal map in style:Layer implementations

### PR DESCRIPTION
**Pull request by pozdnyakov**
Use `mapbox::eternal::hash_map` in `...Layer::setPaintProperty()` and  `...Layer::setLayoutProperty()` methods to decrease code size, binary size and improve performance.

### CODE REVIEW

**Comment by kkaefer** [comment]
\0/
- iOS Dynamic — -81.8 kB
- Android AAR — -16 kB

File: cmake/core.cmake [hunk 1]
```diff
@@ -23,6 +22,7 @@ target_link_libraries(mbgl-core PRIVATE
 # libraries in our public interface.
 target_link_libraries(mbgl-core PUBLIC
     boost
+    eternal
```
Comment (kkaefer) [inline cmake/core.cmake]: Why is this public? It looks like it's only used in the .cpp files.

Comment (pozdnyakov) [reply cmake/core.cmake]: otherwise the `test` target does not compile :/

Comment (kkaefer) [reply cmake/core.cmake]: ...That sounds odd; it doesn't look like there are any test files referencing eternal or headers that include it?

Comment (pozdnyakov) [reply cmake/core.cmake]: Indeed, it was a leftover in `test/style/source.test.cpp` Thanks for the catch!

**Pull request merged**